apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimTransformerV2.java

```diff
@@ -112,132 +198,267 @@ private static CodeableConcept getPriority() {
-                            new CanonicalType(
-                                "http://hl7.org/fhir/us/carin-bb/StructureDefinition/C4BB-Organization")))));
-  }
-
-  private static Reference getPatient(PreAdjFissClaim claimGroup) {
-    return new Reference()
-        .setIdentifier(
+                                    CommonCodings.MC.getSystem(),
+                                    CommonCodings.MC.getCode(),
+                                    CommonCodings.MC.getDisplay())))
+                        .setSystem(
+                            TransformerConstants.CODING_BBAPI_MEDICARE_BENEFICIARY_ID_UNHASHED)
+                        .setValue(claimGroup.getMbi())));
+    patient.setId("patient");
+
+    if (optional.isPresent()) {
+      PreAdjFissPayer benePayer = optional.get();
+      patient
+          .setName(getBeneName(benePayer))
+          .setBirthDate(localDateToDate(benePayer.getBeneDob()))
+          .setGender(
+              benePayer.getBeneSex() == null
+                  ? null
+                  : GENDER_MAP.get(benePayer.getBeneSex().toLowerCase()));
+    }
+
+    return patient;
+  }
+
+  private static List<HumanName> getBeneName(PreAdjFissPayer benePayer) {
+    HumanName name = new HumanName().setFamily(benePayer.getBeneLastName());
+
+    if (benePayer.getBeneFirstName() != null || benePayer.getBeneMidInit() != null) {
+      name.setGiven(
+          List.of(
+              new StringType(benePayer.getBeneFirstName()),
+              new StringType(
+                  benePayer.getBeneMidInit() == null ? null : benePayer.getBeneMidInit() + ".")));
+    }
+
+    return List.of(name);
+  }
+
+  private static Resource getContainedProvider(PreAdjFissClaim claimGroup) {
+    Organization organization = new Organization();
+
+    if (claimGroup.getMedaProvId() != null) {
+      organization
+          .getIdentifier()
+          .add(
+              new Identifier()
+                  .setType(
+                      new CodeableConcept(
+                          new Coding(
+                              C4BBOrganizationIdentifierType.PRN.getSystem(),
+                              C4BBOrganizationIdentifierType.PRN.toCode(),
+                              C4BBOrganizationIdentifierType.PRN.getDisplay())))
+                  .setSystem(BBCodingSystems.PROVIDER_NUM)
+                  .setValue(claimGroup.getMedaProvId()));
+    }
+
+    if (claimGroup.getFedTaxNumber() != null) {
+      organization
+          .getIdentifier()
+          .add(
+              new Identifier()
+                  .setType(
+                      new CodeableConcept(
+                          new Coding(
+                              C4BBOrganizationIdentifierType.TAX.getSystem(),
+                              C4BBOrganizationIdentifierType.TAX.toCode(),
+                              C4BBOrganizationIdentifierType.TAX.getDisplay())))
+                  .setSystem(BBCodingSystems.TAX_NUM)
```

Comment:
We'll need a new system for this, https://bluebutton.cms.gov/resources/variables/fiss/fed-tax-nb